src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -322,19 +322,20 @@ SegmentCleaner::gc_reclaim_space_ret SegmentCleaner::gc_reclaim_space()
 {
   if (!scan_cursor) {
     paddr_t next = P_ADDR_NULL;
-    next.segment = get_next_gc_target();
+    auto seg_addr = next.as_seg_paddr();
```

Comment:
This is invoking the seg_paddr_t copy constructor.  We don't actually want to be creating seg_paddr_t instances, so you'll want to disable the constructors.